pkg/matcher/annotation_tasks_install.go

```diff
@@ -32,31 +34,62 @@ type RemoteTasks struct {
 	Logger            *zap.SugaredLogger
 }
 
-func (rt RemoteTasks) convertToPipeline(data string) (*tektonv1beta1.Pipeline, error) {
+// nolint: dupl
+func (rt RemoteTasks) convertToPipeline(ctx context.Context, data string) (*tektonv1.Pipeline, error) {
 	decoder := k8scheme.Codecs.UniversalDeserializer()
 	obj, _, err := decoder.Decode([]byte(data), nil, nil)
 	if err != nil {
 		return nil, fmt.Errorf("we have a pipeline that is not looking like a kubernetes resource: pipeline: %s resource: %w", data, err)
 	}
 
-	pipeline, ok := obj.(*tektonv1beta1.Pipeline)
-	if !ok {
-		return nil, fmt.Errorf("this doesn't seem to be a proper pipeline")
+	var pipeline *tektonv1.Pipeline
+	switch o := obj.(type) {
+	case *tektonv1.Pipeline:
+		pipeline = o
+	case *tektonv1beta1.Pipeline:
+		c := &tektonv1.Pipeline{}
+		o.SetDefaults(ctx)
+		ctx2 := features.SetFeatureFlag(context.Background())
```

Comment:
Any reason to use `context.Background` instead of `ctx` here ? (you can still use `ctx` later on unchanged).

Reply:
we are not necessary running on cluster with tkn pac resolve... but does it matter to get the right configuration?, since its only for validation, and would further fail when applying on cluster if they don't have that setting

Reply:
didn't want to set this in the normal context we use.... probably going to migrate to use that pattern from pipeline in pac in the feature, which mean removing the struct we pass around and instead properly pass it in the context, gonna need a lot of refactoring for that..

Reply:
`ctx2 := features.SetFeatureFlag(ctx)` <- this won't change the `ctx` reference though but create a new instance. It's a nit though.

Reply:
yeah we only use it for checking and validation, we don't use it anywhere else, which is what I wanted.

Reply:
I am trying to think if there is a use case where we will need to behave differently depending of the tekton configuration but i don't think there is any